pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -310,7 +312,7 @@ func (e *TokensController) syncSecret() {
 		case sa == nil:
 			// Delete token
 			klog.V(4).Infof("syncSecret(%s|%s/%s), service account does not exist, deleting token", secretInfo.clusterName, secretInfo.namespace, secretInfo.name)
-			if retriable, err := e.deleteToken(secretInfo.clusterName, secretInfo.namespace, secretInfo.name, secretInfo.uid); err != nil {
+			if retriable, err := e.deleteToken(logicalcluster.From(secret), secretInfo.namespace, secretInfo.name, secretInfo.uid); err != nil {
```

Comment:
Maybe revert this change, since secretIInfo.clusterName is already what you want?